volatility3/framework/automagic/symbol_finder.py

```diff
@@ -160,8 +160,29 @@ def _banner_scan(
                     path_join(config_path, requirement.name, "symbol_mask")
                 ] = layer.address_mask
 
+                # Keep track of the existing table names so we know which ones were added
+                old_table_names = set(context.symbol_space._dict)
+
                 # Construct the appropriate symbol table
                 requirement.construct(context, config_path)
+
+                new_table_names = context.symbol_space._dict.keys() - old_table_names
+                # It should add only one symbol table. Ignore the next steps if it doesn't
+                if len(new_table_names) == 1:
+                    new_table_name = new_table_names.pop()
+                    symbol_table = context.symbol_space._dict[new_table_name]
+                    producer = symbol_table.producer
+                    vollog.debug(
+                        f"producer_name: {producer.name}, producer_version: {producer.version_string}"
+                    )
+                    for category in symbol_table.metadata._json_data:
```

Comment:
We shouldn't be accessing private members.  It looks like you already added support for returning an appropriate `Metadata` object for Mac.  If there's specific things you need from the linux metadata, they should be expose by the various `*Metadata` classes.

They also won't have been part of the schema, so you'll need to write them defensively to return no data if they're not populated, but not to fail (and possibly to update the schema, so things like `dwarf2json` or `btf2json` know that they need to populate...

Reply:
done